### PR TITLE
Add a binder link to the top of notebooks in the docs

### LINKED ISSUE #3083: Add binder links to top of all docs pages
### Problem

Parts of the docs make far more sense when there is a working python kernel. Also it's nice to be able to play around with the widgets. So the semi-static versions on readthedocs are non-optimal

### Proposed Solution

Use an nbsphinx prolog to generate a binder link at the top of every notebook

https://nbsphinx.readthedocs.io/en/0.3.0/prolog-and-epilog.html

### Additional context

here's an implementation of this that I did:
https://github.com/ianhi/mpl-interactions/blob/e2b2a87e84de3087abf5552b8adbff6dc0cf9bb0/docs/conf.py#L68-L87

and the result 
(https://mpl-interactions.readthedocs.io/en/stable/examples/Usage-Guide.html)
image

### PR DESCRIPTION
**Pull request by ianhi**
Closes: https://github.com/jupyter-widgets/ipywidgets/issues/3083

Following: https://github.com/spatialaudio/nbsphinx/blob/98005a9d6b331b7d6d14221539154df69f7ae51a/doc/conf.py#L38

Example on this page:
https://nbsphinx.readthedocs.io/en/0.8.2/prolog-and-epilog.html

### CODE REVIEW

**Comment by ianhi** [comment]
Example here:
https://ipywidgets--3164.org.readthedocs.build/en/3164/examples/Widget%20List.html

image

**Comment by ianhi** [comment]
Oh dang the spaces in the notebook names are wreaking absolute havoc

**Comment by ianhi** [comment]
> Oh dang the spaces in the notebook names are wreaking absolute havoc

Which semirelatedly led to discovering https://github.com/jupyterlab/jupyterlab/issues/9953

**Comment by ianhi** [comment]
That seems to have fixed both the links. I also added a new line between the two statements as otherwise the binder badge was often left all on its own on the second line which looked lonely.

File: docs/source/conf.py [hunk 1]
```diff
@@ -29,6 +29,28 @@
     'jupyter': ('https://jupyter.readthedocs.io/en/latest/', None),
 }
 
+
+# prolog taken based on https://github.com/spatialaudio/nbsphinx/blob/98005a9d6b331b7d6d14221539154df69f7ae51a/doc/conf.py#L38
```
Comment (jtpio) [inline docs/source/conf.py:33]: Minor wording tweak:

```suggestion
# prolog based on https://github.com/spatialaudio/nbsphinx/blob/98005a9d6b331b7d6d14221539154df69f7ae51a/doc/conf.py#L38
```

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**